test/Microsoft.AspNetCore.Server.KestrelTests/KestrelServerTests.cs

```diff
@@ -55,6 +56,23 @@ public void StartWithEmptyAddressesThrows()
             Assert.Equal(1, testLogger.CriticalErrorsLogged);
         }
 
+        [Fact]
+        public void StartWarnsWhenIgnoringIServerAddressesFeature()
+        {
+            var testLogger = new TestApplicationErrorLogger();
+            var options = new KestrelServerOptions();
+
+            // Directly configuring an endpoint using Listen, causes the IServerAddressesFeature to be ignored.
+            options.Listen(IPAddress.Loopback, 0);
+
+            using (var server = CreateServer(options, testLogger))
+            {
+                server.Features.Get<IServerAddressesFeature>().Addresses.Add("http://localhost:5001");
+                StartDummyApplication(server);
+                Assert.Equal(1, testLogger.TotalWarningsLogged);
```

Comment:
Check actual message. Also add more addresses to the feature (or better, make it a theory and have cases for single and multiple addresses).